## Assignment 01 

Create code to do the following:
<font color='red'>**(Use SHELL SCRIPT code ONLY - DO NOT use python):**</font>

**1.** Using the data file `"data.csv"`, create a new version of the data in a new file `data_v2.csv` that contains only the following columns: `customerID,gender,PhoneService,InternetService,StreamingTV,Contract,TotalCharges,Churn`

In [1]:
%%!
# Write your code Here
cut -d',' -f1,2,7,9,14,16,20,21 data.csv > data_v2.txt
cat data_v2.txt

['customerID,gender,PhoneService,InternetService,StreamingTV,Contract,TotalCharges,Churn',
 '7590-VHVEG,Female,No,DSL,No,Month-to-month,29.85,No',
 '5575-GNVDE,Male,Yes,DSL,No,One year,1889.5,No',
 '3668-QPYBK,Male,Yes,DSL,No,Month-to-month,108.15,Yes',
 '7795-CFOCW,Male,No,DSL,No,One year,1840.75,No',
 '9237-HQITU,Female,Yes,Fiber optic,No,Month-to-month,151.65,Yes',
 '9305-CDSKC,Female,Yes,Fiber optic,Yes,Month-to-month,820.5,Yes',
 '1452-KIOVK,Male,Yes,Fiber optic,Yes,Month-to-month,1949.4,No',
 '6713-OKOMC,Female,No,DSL,No,Month-to-month,301.9,No',
 '7892-POOKP,Female,Yes,Fiber optic,Yes,Month-to-month,3046.05,Yes',
 '6388-TABGU,Male,Yes,DSL,No,One year,3487.95,No',
 '9763-GRSKD,Male,Yes,DSL,No,Month-to-month,587.45,No',
 '7469-LKBCI,Male,Yes,No,No internet service,Two year,326.8,No',
 '8091-TTVAX,Male,Yes,Fiber optic,Yes,One year,5681.1,No',
 '0280-XJGEX,Male,Yes,Fiber optic,Yes,Month-to-month,5036.3,Yes',
 '5129-JLPIS,Male,Yes,Fiber optic,Yes,Month-to-month,2686.05,No',
 '3655-SN

**2.** From the data file `"data_v2.csv"` Keep only those who have phoneService with StreamingTV (save result in a new file called `data_v3.csv`

**Hints:**
- The default separator between values is "space, tab and newLine sign" i.e. `' \t\n'`. In order to specify the default separator between the values to be newLine sign in your system, you can use `IFS=$'\n'` before you run your code, then set that back to be: `IFS=$' \t\n'` at the end of your code.
- if you had a variable such as `x` and you want to pass its value to another instruction suh as `cut`, and save results in variable `y`. then you can do that as follows:
`y=$(echo $x | cut)`

In [2]:
%%!
IFS=$'\n'
# Write your code Here
while read line
do
    x=$(echo "$line" | cut -d',' -f3)
    y=$(echo "$line" | cut -d',' -f5)
    if [ $x == "Yes" -a $y == "Yes" ]
    then
                echo $line
    fi       
    done < data_v2.csv > data_v3.csv
# Your code ends here
cat data_v3.csv
IFS=$' \t\n'

['9305-CDSKC,Female,Yes,Fiber optic,Yes,Month-to-month,820.5,Yes',
 '1452-KIOVK,Male,Yes,Fiber optic,Yes,Month-to-month,1949.4,No',
 '7892-POOKP,Female,Yes,Fiber optic,Yes,Month-to-month,3046.05,Yes',
 '8091-TTVAX,Male,Yes,Fiber optic,Yes,One year,5681.1,No',
 '0280-XJGEX,Male,Yes,Fiber optic,Yes,Month-to-month,5036.3,Yes',
 '5129-JLPIS,Male,Yes,Fiber optic,Yes,Month-to-month,2686.05,No',
 '3655-SNQYZ,Female,Yes,Fiber optic,Yes,Two year,7895.15,No',
 '9959-WOFKT,Male,Yes,Fiber optic,Yes,Two year,7382.25,No',
 '6467-CHFZW,Male,Yes,Fiber optic,Yes,Month-to-month,4749.15,Yes',
 '5248-YGIJN,Male,Yes,DSL,Yes,Two year,6369.45,No',
 '8773-HHUOZ,Female,Yes,DSL,Yes,Month-to-month,1093.1,Yes',
 '4929-XIHVW,Male,Yes,Fiber optic,Yes,Month-to-month,181.65,No',
 '6234-RAAPL,Female,Yes,Fiber optic,Yes,Two year,7251.7,No',
 '5380-WJKOV,Male,Yes,Fiber optic,Yes,Month-to-month,3549.25,Yes',
 '8168-UQWWF,Female,Yes,Fiber optic,Yes,Month-to-month,1105.4,Yes',
 '9489-DEDVP,Female,Yes,DSL,Yes,Two year,4872.

**3.** Using the data file `"data_v3.csv"`, find the contract type that has the largest number clients

In [3]:
%%!
# Write your code Here
cut -d',' -f6 data_v3.csv > contract.txt
sort contract.txt | uniq -c
#cat data_v3.csv|cut -d, -f6|sort|uniq -c|sort -n -r|head -1

['   1178 Month-to-month', '    582 One year', '    662 Two year']

Therefore, we can see that 'Month-to-Month' contract has the largest number of clients. 

**4.** Using the data file `"data_v3.csv"`, Find the number of Female vs male in each contract type

In [4]:
%%!
# Write your code Here
grep 'Month-to-month' data_v3.csv > mtm.csv
cut -d',' -f2 mtm.csv > mtm2.csv
sort mtm2.csv | uniq -c

['    580 Female', '    598 Male']

In [5]:
%%!
grep 'Two year' data_v3.csv > twoy.csv
cut -d',' -f2 twoy.csv > twoy2.csv
sort twoy2.csv | uniq -c

['    336 Female', '    326 Male']

In [6]:
%%!
grep 'One year' data_v3.csv > oney.csv
cut -d',' -f2 oney.csv > oney2.csv
sort oney2.csv | uniq -c

['    299 Female', '    283 Male']

We can see that there are 580 females and 598 males in month-to-month contract, 336 females ad 326 males in two year contract, and 299 females and 283 males in one year contract. 

**5.** Using the data file `"data_v3.csv"`, Create three separate files (one for each type of contracts):

- `Month-to-month.csv` that contains only those with Month-to-month contract type
- `Two year.csv` that contains only those with Two years contract type
- `One year.csv` that contains only those with One year contract type

In [7]:
%%!
# Write your code Here
grep 'Month-to-month' data_v3.csv > Month-to-month.csv
grep 'Two year' data_v3.csv > 'Two year.csv'
grep 'One year' data_v3.csv > 'One year.csv'
cat 'Month-to-month.csv'

['9305-CDSKC,Female,Yes,Fiber optic,Yes,Month-to-month,820.5,Yes',
 '1452-KIOVK,Male,Yes,Fiber optic,Yes,Month-to-month,1949.4,No',
 '7892-POOKP,Female,Yes,Fiber optic,Yes,Month-to-month,3046.05,Yes',
 '0280-XJGEX,Male,Yes,Fiber optic,Yes,Month-to-month,5036.3,Yes',
 '5129-JLPIS,Male,Yes,Fiber optic,Yes,Month-to-month,2686.05,No',
 '6467-CHFZW,Male,Yes,Fiber optic,Yes,Month-to-month,4749.15,Yes',
 '8773-HHUOZ,Female,Yes,DSL,Yes,Month-to-month,1093.1,Yes',
 '4929-XIHVW,Male,Yes,Fiber optic,Yes,Month-to-month,181.65,No',
 '5380-WJKOV,Male,Yes,Fiber optic,Yes,Month-to-month,3549.25,Yes',
 '8168-UQWWF,Female,Yes,Fiber optic,Yes,Month-to-month,1105.4,Yes',
 '4080-IIARD,Female,Yes,DSL,Yes,Month-to-month,981.45,No',
 '7760-OYPDY,Female,Yes,Fiber optic,Yes,Month-to-month,144.15,Yes',
 '8012-SOUDQ,Female,Yes,Fiber optic,Yes,Month-to-month,3838.75,No',
 '9420-LOJKX,Female,Yes,Fiber optic,Yes,Month-to-month,1426.4,Yes',
 '6575-SUVOI,Female,Yes,DSL,Yes,Month-to-month,1752.65,No',
 '1658-BYGOY,Male

In [8]:
cat 'Two year.csv'






















































































































































































































































































































































































































































































































































































































































































3186-AJIEK,Male,Yes,Fiber optic,Yes,Two year,6844.5,No


In [9]:
cat 'One year.csv'